pkg/mover/chart.go

```diff
@@ -1,90 +1,163 @@
-package mover
-
 // Copyright 2021 VMware, Inc.
 // SPDX-License-Identifier: BSD-2-Clause
 
+package mover
+
 import (
+	"errors"
 	"fmt"
 	"io/ioutil"
 	"os"
 
 	"github.com/avast/retry-go"
 	"github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/v2/internal"
-	"github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/v2/pkg/rewrite"
 	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
 	"helm.sh/helm/v3/pkg/chartutil"
 )
 
-const DefaultRetries = 3
+const (
+	defaultRetries        = 3
+	EmbeddedHintsFilename = ".relok8s-images.yaml"
+)
+
+var ErrImageHintsMissing = errors.New("no image hints provided`")
 
 type Logger interface {
 	Printf(format string, i ...interface{})
+	Println(i ...interface{})
+}
+
+type defaultLogger struct{}
+
+func (l *defaultLogger) Printf(format string, i ...interface{}) {
+	fmt.Printf(format, i...)
+	return
+}
+
+func (l *defaultLogger) Println(i ...interface{}) {
+	fmt.Println(i...)
+	return
 }
 
 type ChartMover struct {
-	chart        *chart.Chart
+	chart *chart.Chart
+	// Extracted metadata from the provided Helm Chart
+	ChartName    string
+	ChartVersion string
```

Comment:
Why these need to be public?

Reply:
Good question. It's because now the Chart object is an implementation detail of the library, not from the command. That way a consumer of the library does not need to import the Helm library and so on. 

But by doing so, some metadata (name and version) that is currently used to calculate the resulting chart filename will be out of reach. These two exposed properties are used to still fulfill that requirement. Note that I could have exposed the Chart itself but again, I'd prefer not to since to me loader.Chart is an implementation detail that we even might swap in the future.